pkg/libinit/root_freebsd.go

```diff
@@ -0,0 +1,393 @@
+// Copyright 2014-2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package libinit creates the environment and root file system for u-root.
+package libinit
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"github.com/u-root/u-root/pkg/cmdline"
+	"github.com/u-root/u-root/pkg/cp"
+	"github.com/u-root/u-root/pkg/ulog"
+	"golang.org/x/sys/unix"
+)
+
+type Creator interface {
+	Create() error
+	fmt.Stringer
+}
+
+type Dir struct {
+	Name string
+	Mode os.FileMode
+}
+
+func (d Dir) Create() error {
+	return os.MkdirAll(d.Name, d.Mode)
+}
+
+func (d Dir) String() string {
+	return fmt.Sprintf("dir %q (mode %#o)", d.Name, d.Mode)
+}
+
+type Symlink struct {
+	Target  string
+	NewPath string
+}
+
+func (s Symlink) Create() error {
+	os.Remove(s.NewPath)
+	return os.Symlink(s.Target, s.NewPath)
+}
+
+func (s Symlink) String() string {
+	return fmt.Sprintf("symlink %q -> %q", s.NewPath, s.Target)
+}
+
+type Dev struct {
+	Name string
+	Mode uint32
+	Dev  uint64
+}
+
+func (d Dev) Create() error {
+	os.Remove(d.Name)
+	return unix.Mknod(d.Name, d.Mode, d.Dev)
+}
+
+func (d Dev) String() string {
+	return fmt.Sprintf("dev %q (mode %#o; magic %d)", d.Name, d.Mode, d.Dev)
+}
+
+type Mount struct {
+	Source string
+	Target string
+	FSType string
+	Flags  uintptr
+	Opts   string
+}
+
+func (m Mount) Create() error {
+	return os.ErrInvalid
```

Comment:
We have a lot of Mounts in []Namespace slice and all of them will return an error. There is a plan to rewrite Namespaces slice in future?

Reply:
I need to fix mount, but it got complicated, so I wanted to let it go for the next set of PRs. The main goal of this series is to at least get freebsd in the CI testing.